src/utils/llm_endpoint.py

```diff
@@ -0,0 +1,51 @@
+import os
+from dataclasses import dataclass
+from openai import OpenAI
+from requests.exceptions import RequestException
+
+@dataclass
+class LLM:
+    platform: str
+    model_name: str
+    key: str = ""
+
+    def __post_init__(self):
+        if self.platform == 'openai':
+            if not self.key:
+                self.key = os.getenv('OPENAI_API_KEY')
+                if not self.key:
+                    raise RuntimeError("'OPENAI_API_KEY' not found via os.getenv" )
+            self.client = OpenAI(
+                api_key=self.key
+            )
+
+        elif self.platform == 'openrouter':
+            if not self.key:
+                self.key = os.getenv('OPENROUTER_API_KEY')
+                if not self.key:
+                    raise RuntimeError("'OPENROUTER_API_KEY' not found via os.getenv" )
+            self.client = OpenAI(
+                base_url="https://openrouter.ai/api/v1",
+                api_key=self.key,
+            )
+        elif self.platform == 'ollama':
+            self.client = OpenAI(
```

Comment:
Let's link the ref https://github.com/ollama/ollama/blob/main/docs/openai.md for future readers